src/mp/util.cpp

```diff
@@ -37,14 +39,17 @@ size_t MaxFd()
 
 std::string ThreadName(const char* exe_name)
 {
-    char thread_name[17] = {0};
+    char thread_name[16] = {0};
+#ifdef HAVE_PTHREAD_GETNAME_NP
     pthread_getname_np(pthread_self(), thread_name, sizeof(thread_name));
-    uint64_t tid = 0;
-#if __linux__
-    tid = syscall(SYS_gettid);
```

Comment:
I'd like keep using `gettid` on linux if possible. On linux, tid thread numbers are different from pthread numbers, and match up with the thread numbers shown in gdb, so it's easier to match up logs with gdb backtraces. Tid numbers are also shorter by default and easier to visually distinguish (`20080` vs `140369161336576`)

Reply:
I see. Done. And also resurrected the other call to `pthread_threadid_np()` (+ a proper check for its presence).

I checked that on FreeBSD `std::this_thread::get_id()` returns some long id that is different from what gdb prints (similar to what you say for linux), but [`thr_self()`](https://www.freebsd.org/cgi/man.cgi?query=thr_self) and [`pthread_getthreadid_np()`](https://www.freebsd.org/cgi/man.cgi?query=pthread_getthreadid_np) print the same id as gdb. So I added `pthread_getthreadid_np()` too, and if none is available then fallback to `std::this_thread::get_id()`.